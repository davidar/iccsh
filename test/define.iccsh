:define LIFE (40+2)
:printf %d LIFE

:define
PRINT3(...)
    printf("%d %d %d\n", __VA_ARGS__)

:eval PRINT3(0, 4, 2)

:define STRINGIFY(arg) #arg
:printf %s STRINGIFY(1+2=3)

:define STRINGIFY_MACRO(arg) STRINGIFY(arg)
:printf %s STRINGIFY_MACRO(LIFE)

:undefine LIFE
:printf %s STRINGIFY_MACRO(LIFE)
