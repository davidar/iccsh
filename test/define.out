42
0 4 2
1+2=3
(40+2)
LIFE
